pkg/generate/app/pipeline.go

```diff
@@ -28,21 +35,76 @@ type Pipeline struct {
 	Labels     map[string]string
 }
 
-// NewImagePipeline creates a new pipeline with components that are not expected
-// to be built.
-func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
-	return &Pipeline{
-		From:  from,
-		Image: image,
-	}, nil
+// NewPipeline creates a new pipeline from a component reference.
+func NewPipeline(name string, names map[string]int, ref ComponentReference, environment Environment, addEnvToBuild bool, labels map[string]string, outputDocker bool, deploy bool) (*Pipeline, error) {
```

Comment:
> I don't like passing component reference into here.

@smarterclayton I've iterated on it once again and was able to get rid of passing the ref.